Fix World Info Entry Priority Sort

KhreedAI opened the pull request:
Corrected and expanded priority sort logic  for sortWorldInfoEntries to sort by:

First active (not disabled), then constant, then normal, then disabled, finally by comment.
Additional steps to ensure disabled entries follow the same pattern.

- Verified that active entries consistently appear before inactive ones and that the sorting updates correctly.

<!-- Put X in the box below to confirm -->
- [X ] I have read the [Contributing guidelines](https://github.com/SillyTavern/SillyTavern/blob/release/CONTRIBUTING.md).

KhreedAI committed:
Fix Priority Sort

Corrected and expanded priority sort logic to sort by:

First constant, then normal, then disabled, then by comment.

Cohee1207 commented:
Can you please explain on this change? Using the comment field for the priority would not reflect the internal process of checking for entries inclusion. Also, we are already using order and uid as tie-breakers within each group.

KhreedAI committed:
Re-evaluate world info entry - simplified priority sort order

Changed priority primarySort ternary to check disabled first, then constant, then normal.

KhreedAI commented:
I've simplified the PR by adjusting only the order of operations, while keeping the existing logic for secondary and tertiary sorting intact.

Now, by checking for disabled entries first, all disabled items are placed at the bottom.

Before:  (a.constant ? 0 : a.disable ? 2 : 1;)
![current_sort](https://github.com/user-attachments/assets/d0aac528-05d9-4f20-8b4d-56d536b3815c)


After: (a.disable ? 2 : a.constant ? 0 : 1)
![simple_corrected_sort](https://github.com/user-attachments/assets/0d45ea2f-a5a8-4c43-b917-48d05a6b9549)

Wolfsblvt reviewed (approved):
I was just typing a very long response why your change didn’t make any sense. Even after seeing your commit.
But holy fuck, you are right.

@Cohee1207 this is an oversight when we switched WI back to having a separate disabled/enabled toggle. Because now entries can be both declared as constant, but also currently disabled.

Makes sense this was, easiest fix to just check disable first.
Now the *i-Tüpfelchen* would be if we could sort disabled but also constant entries above the disabled non-constant entries, but I am not sure if that matters too much.

I approve this. Good catch.

Cohee1207 reviewed (approved):
Thanks for the explanation, that totally makes sense now.

KhreedAI commented:
> "Now the i-Tüpfelchen would be if we could sort disabled but also constant entries above the disabled non-constant entries, but I am not sure if that matters too much."

@Wolfsblvt  This could potentially be done by adding a new quaternarySort to the sorting composite, allowing us to remove the disable check from the 'priority' sort.

// Push disabled items to the bottom
```
    const quaternarySort = (a, b) => (a.disable ? 1 : 0) - (b.disable ? 1 : 0);
    ...
        data.sort((a, b) => {
        return quaternarySort(a,b) || primarySort(a, b) || secondarySort(a, b) || tertiarySort(a, b);
    });
```
Putting quaternarySort(a,b) first in data.sort would universally place disabled items at the bottom for every sorting method.

Cohee1207 commented:
> This could potentially be done by adding a new quaternarySort to the sorting composite, allowing us to remove the disable check from the 'priority' sort.

I think it is fine as it is for now. Good to merge.
